File: mobile/src/main/java/com/alexstyl/specialdates/upcoming/ui/UpcomingRowViewHolder.java
package com.alexstyl.specialdates.upcoming.ui;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.alexstyl.specialdates.upcoming.view.OnUpcomingEventClickedListener;

abstract class UpcomingRowViewHolder<T> extends RecyclerView.ViewHolder {

    UpcomingRowViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bind(T element, OnUpcomingEventClickedListener listener);
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/upcoming/UpcomingRowViewHolder.kt
package com.alexstyl.specialdates.upcoming

import android.support.v7.widget.RecyclerView
import android.view.View

import com.alexstyl.specialdates.upcoming.view.OnUpcomingEventClickedListener

abstract class UpcomingRowViewHolder<in T : UpcomingRowViewModel>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    abstract fun bind(viewModel: T, listener: OnUpcomingEventClickedListener)
}
